proxy.go

```diff
@@ -617,3 +624,13 @@ func (p *Proxy) connect(req *http.Request) (*http.Response, net.Conn, error) {
 
 	return proxyutil.NewResponse(200, nil, req), conn, nil
 }
+
+func (p *Proxy) SetOnClosedConnectionError(cb func(gocontext.Context, string, error)) {
+	p.onClosedConnectionError = cb
+}
+
+func (p *Proxy) OnClosedConnectionError(ctx gocontext.Context, serverName string, err error) {
+	if p.onClosedConnectionError != nil {
+		p.onClosedConnectionError(ctx, serverName, err)
+	}
```

Comment:
this function should not be exposed to the public API, we should just do this if check where the internal function is called